src/anomalib/metrics/per_image/binclf_curve_numpy.py

```diff
@@ -0,0 +1,430 @@
+
+
+def binclf_multiple_curves(
+    scores_batch: ndarray,
+    gts_batch: ndarray,
+    threshs: ndarray,
+    algorithm: BinclfAlgorithm | str = BinclfAlgorithm.NUMBA.value,
+) -> ndarray:
+    """Multiple binary classification matrix (per-instance scope) at each threshold (shared).
+
+    This is a wrapper around `_binclf_multiple_curves_python` and `_binclf_multiple_curves_numba`.
+    Validation of the arguments is done here (not in the actual implementation functions).
+
+    Note: predicted as positive condition is `score >= thresh`.
+
+    Args:
+        scores_batch (ndarray): Anomaly scores (N, D,).
+        gts_batch (ndarray): Binary (bool) ground truth of shape (N, D,).
+        threshs (ndarray): Sequence of thresholds in ascending order (K,).
+        algorithm (str, optional): Algorithm to use. Defaults to ALGORITHM_NUMBA.
+
+    Returns:
+        ndarray: Binary classification matrix curves (N, K, 2, 2)
+
+        The last two dimensions are the confusion matrix (ground truth, predictions)
+        So for each thresh it gives:
+            - `tp`: `[... , 1, 1]`
+            - `fp`: `[... , 0, 1]`
+            - `fn`: `[... , 1, 0]`
+            - `tn`: `[... , 0, 0]`
+
+        `t` is for `true` and `f` is for `false`, `p` is for `positive` and `n` is for `negative`, so:
+            - `tp` stands for `true positive`
+            - `fp` stands for `false positive`
+            - `fn` stands for `false negative`
+            - `tn` stands for `true negative`
+
+        The numbers in each confusion matrix are the counts (not the ratios).
+
+        Counts are relative to each instance (i.e. from 0 to D, e.g. the total is the number of pixels in the image).
+
+        Thresholds are shared across all instances, so all confusion matrices, for instance,
+        at position [:, 0, :, :] are relative to the 1st threshold in `threshs`.
+
+        Thresholds are sorted in ascending order.
+    """
+    algorithm = BinclfAlgorithm(algorithm)
+    _validate_is_scores_batch(scores_batch)
+    _validate_is_gts_batch(gts_batch)
+    _validate.is_same_shape(scores_batch, gts_batch)
+    _validate.is_threshs(threshs)
+
+    if algorithm == BinclfAlgorithm.NUMBA:
+        if HAS_NUMBA:
+            return _binclf_curve_numba.binclf_multiple_curves_numba(scores_batch, gts_batch, threshs)
+
+        logger.warning(
+            f"Algorithm '{BinclfAlgorithm.NUMBA.value}' was selected, but Numba is not installed. "
+            f"Falling back to '{BinclfAlgorithm.PYTHON.value}' implementation.",
+            "Notice that the performance will be slower. Consider installing Numba for faster computation.",
+        )
+
+    return _binclf_multiple_curves_python(scores_batch, gts_batch, threshs)
+
+
+# ========================================= PER-IMAGE BINCLF CURVE =========================================
+
+
+def _get_threshs_minmax_linspace(anomaly_maps: ndarray, num_threshs: int) -> ndarray:
```

Comment:
Not sure if using threshs over thresholds would massively reduce the verbosity. As a reader, i would prefer the full for clarity.

```suggestion
def _get_threshs_minmax_linspace(anomaly_maps: ndarray, num_thresholds: int) -> ndarray:
```